modules/core/lib/utils/network_manager/no_network_wiget.dart

```diff
@@ -0,0 +1,79 @@
+import 'package:core/utils/network_manager/network_change_manager.dart';
+import 'package:flutter/material.dart';
+
+class NoNetworkWidget extends StatefulWidget {
+  const NoNetworkWidget({super.key});
+
+  @override
+  State<NoNetworkWidget> createState() => _NoNetworkWidgetState();
+}
+
+class _NoNetworkWidgetState extends State<NoNetworkWidget> with StateMixin {
+  late final INetworkChangeManager _networkChange;
+  NetworkResult? _networkResult;
+  @override
+  void initState() {
+    super.initState();
+    _networkChange = NetworkChangeManager.instance;
+    fetchFirstResult();
+
+    waitForScreen(() {
+      _networkChange.handleNetworkChange((result) {
+        _updateView(result);
+      });
+    });
+  }
+
+  // when app start, check network connection because it could start with no connection
+  Future<void> fetchFirstResult() async {
+    waitForScreen(() async {
+      final result = await _networkChange.checkNetworkOneTime();
+      _updateView(result);
+    });
+  }
+
+  void _updateView(NetworkResult result) {
+    setState(() {
+      _networkResult = result;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    // TODO: MAKE GLOBAL SIZES
+    Size size = MediaQuery.of(context).size;
+    return Material(
+      color: Colors.transparent,
+      child: AnimatedCrossFade(
+        duration: const Duration(milliseconds: 500),
+        crossFadeState: _networkResult == NetworkResult.off
+            ? CrossFadeState.showFirst
+            : CrossFadeState.showSecond,
+        firstChild: Container(
+          height: size.height * .08,
+          color: Colors.black,
+          child: Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: const [
+              Text(
+                "Please Check Internet Your Connection !",
+                style: TextStyle(color: Colors.white),
+                overflow: TextOverflow.visible,
+                textAlign: TextAlign.center,
+              ),
+            ],
+          ),
+        ),
+        secondChild: const SizedBox(),
+      ),
+    );
+  }
+}
+
```

Comment:
The display of warning messages should be entirely managed by the ConnectivityWidget, and to maintain the layout of the current screen, all of these processes should be carried out using the Stack widget.

Reply:
actually, I don't understand why we should use stack widget.

Reply:
Answer is here https://github.com/SalihCanBinboga/recipe_finder/pull/19#discussion_r1182587898